src/main/kotlin/com/sourcegraph/cody/history/state/HistoryState.kt

```diff
@@ -9,4 +9,6 @@ class HistoryState : BaseState() {
 
   @get:OptionTag(tag = "defaultEnhancedContext", nameAttribute = "")
   var defaultEnhancedContext: EnhancedContextState? by property()
+
+  fun doIncrementModificationCount() = super.incrementModificationCount()
```

Comment:
https://github.com/JetBrains/intellij-community/blob/idea/233.14475.28/platform/projectModel-api/src/com/intellij/openapi/components/BaseState.kt

That file ☝️ mentions that it is our responsibility to call the method.

https://github.com/JetBrains/intellij-community/blob/idea/233.14475.28/platform/projectModel-api/src/com/intellij/openapi/components/PersistentStateComponentWithModificationTracker.java

Instead of above we could just try https://github.com/JetBrains/intellij-community/blob/idea/233.14475.28/platform/projectModel-api/src/com/intellij/openapi/components/PersistentStateComponent.java.

Reply:
@pkukielka, do we have the steps to reproduce the flakiness issue?

Reply:
I managed to reproduce it by running the IDE from `:runIde` (the Gradle task from IntelliJ) and killing the process from IntelliJ. I'd expect the same when we kill the process from Activity Monitor. I do not think those cases are important for us - killing a process extraordinary way to exit the IDE. 

(but maybe tere are more common scenarios to reproduce the issue?)